File: Programmers/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> days;

    for (int i = 0; i < progresses.size(); i++) {
        int remaining = 100 - progresses[i];
        int required_days = (remaining + speeds[i] - 1) / speeds[i];
        days.push(required_days);
    }

    while (!days.empty()) {
        int count = 1; 
        int first = days.front();
        days.pop();

        while (!days.empty() && days.front() <= first) {
            count++;
            days.pop();
        }

        answer.push_back(count);
    }

    return answer;
}

File: KB boot/week2/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> queue = new LinkedList<>();
        int[] tempAnswer = new int[progresses.length];  
        int idx = 0;

        for (int i = 0; i < progresses.length; i++) {
            int remaining = 100 - progresses[i];
            int requiredDays = (remaining + speeds[i] - 1) / speeds[i];
            queue.offer(requiredDays);
        }

        while (!queue.isEmpty()) {
            int first = queue.poll();
            int count = 1;

            while (!queue.isEmpty() && queue.peek() <= first) {
                queue.poll();
                count++;
            }

            tempAnswer[idx++] = count;
        }

        return Arrays.copyOf(tempAnswer, idx);
    }
}
